lib/ohai/mixin/network_helper.rb

```diff
@@ -47,7 +54,27 @@ def hex_to_dec_netmask(netmask)
       # server), and the method should return the hostname and not the IP address.
       #
       def canonicalize_hostname(hostname)
-        Addrinfo.getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME).first.canonname
+        ai = Addrinfo
+          .getaddrinfo(hostname, nil, nil, nil, nil, Socket::AI_CANONNAME)
+          .first
+
+        canonname = ai&.canonname
+        # use canonname if it's an FQDN
+        # This API is preferred as it never gives us an IP address for broken DNS
+        # (see https://github.com/chef/ohai/pull/1705)
+        # However, we have found that Windows hosts that are not joined to a domain
+        # can return a non-qualified hostname).
+        # Use a '.' in the canonname as indicator of FQDN
+        return canonname if /\./.match?(canonname)
```

Comment:
Don't regex when you can include, much cheaper.

```suggestion
        return canonname canonname.include?('.')
```

Reply:
`return canonname if canonname.include?(".")`